internal/evaluator/builder.go

```diff
@@ -266,21 +260,10 @@ func (bldr *Builder) processIgnoreEmpty(
 ) error {
 	// the only time we need to ignore empty on a value is if it's evaluating a
 	// field item (repeated element or map key/value).
-	val.IgnoreEmpty = forItems && (constraints.GetIgnoreEmpty() ||
-		constraints.GetIgnore() == validate.Ignore_IGNORE_IF_UNPOPULATED ||
-		constraints.GetIgnore() == validate.Ignore_IGNORE_IF_DEFAULT_VALUE)
-	switch {
-	case !val.IgnoreEmpty:
-		// only need the zero value for checking ignore_empty constraint
+	val.IgnoreEmpty = forItems && bldr.shouldIgnoreEmpty(constraints)
+	if val.IgnoreEmpty {
+		val.Zero = bldr.zeroValue(fdesc, forItems)
 		return nil
```

Comment:
Can remove this return and fall back to one immediately after.

Reply:
weird that a linter didn't catch that...